app/web/features/search/SearchPage.tsx

```diff
@@ -52,193 +55,195 @@ const useStyles = makeStyles((theme) => ({
   },
 }));
 
-export default function SearchPage() {
+/**
+ * Search page, queries the backend & obtains all the users to be shown on the map, creates all the state variables and sends them to the search page sub-components
+ */
+export default function SearchPage({
+  locationName,
+  bbox,
+}: {
+  locationName: string;
+  bbox: [number, number, number, number];
+}) {
+  const {
+    data: userData,
+    error: errorUser,
+    isLoading: isLoadingUser,
+  } = useCurrentUser();
   const { t } = useTranslation([GLOBAL, SEARCH]);
+  const queryClient = new QueryClient();
   const classes = useStyles();
   const theme = useTheme();
-
   const map = useRef<MaplibreMap>();
+
+  // State
+  const [locationResult, setLocationResult] = useState({
+    bbox: bbox,
+    isRegion: false,
+    location: { lng: undefined, lat: undefined },
+    name: locationName,
+    simplifiedName: locationName,
+  });
+  const [queryName, setQueryName] = useState<undefined | string>(undefined);
+  const [searchType, setSearchType] = useState("location");
+  const [lastActiveFilter, setLastActiveFilter] = useState(0);
+  const [hostingStatusFilter, setHostingStatusFilter] = useState(0);
+  const [numberOfGuestFilter, setNumberOfGuestFilter] = useState(undefined);
+  const [completeProfileFilter, setCompleteProfileFilter] = useState(true);
```

Comment:
You've got types for the first one maybe also good to add types for the rest?

I dunno how serious y'all are about Typescript, every dev team is different haha.

Reply:
true I could type that... for me personally I think it's better to have them rather not (except when they are specially big/complex and make the code dirty)

Reply:
about our team preferences I don't think exist that right now, because it's mainly me working on the frontend side, and the codebase was made by developers who barely work now or not at all, so it's more less up to us... I would say that if the code it's clean/readable/tested/good enough it's fine :D

Reply:
Sweet sounds good. Yeah I agree, even if Typescript is annoying it does often help me find bugs in my code.